replace_with/src/lib.rs

```diff
@@ -0,0 +1,8 @@
+// Julien Chien <jchien17@cmc.edu>
+
+pub fn replace_with<T, F: FnOnce(T) -> T>(t: &mut T, f: F) {
+  unsafe {
+    let mut ptr: *mut T = t;
+    *ptr = f(std::ptr::read(ptr))
```

Comment:
Assigning to `*ptr` doesn't only overwrite the data at `*ptr`, it also drops the original data. Likely what you want here is `ptr::write`.

The provided tests detect this error.